### PR TITLE
Added TestHelpers and service test hierarchy skeleton

### PR DESCRIPTION
**Pull request by wolfe-lienhardt**
This pull request has the base hierarchy for the service tests as well as the AbstractTestHelper class.

These test classes will be added to and fleshed out as more functionality is added to the Services themselves.

The next step is to add the relevant methods for each Service layer, for example: CollectedService would have methods like "FindByFieldWorker" which would take a field worker and return all entities Collected by that field worker. Validators will be added to each layer of the Service hierarchy as well as each entity specific service.

### CODE REVIEW

File: src/test/java/org/openhds/service/AuditableServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    @Before
+    public void setup() throws Exception {
+        initialize(service, helper);
+        sampleDataGenerator.clearData();
+        sampleDataGenerator.generateSampleData();
+    }
+
+    @Test
+    public void findAll(){
+        assertNotNull(service.findAll());
```
Comment (benjamin-heasly) [inline src/test/java/org/openhds/service/AuditableServiceTest.java:46]: Do you think we can also assert that findAll() returns at least one entity?

I guess we'd have to make this a convention of the SampleDataGenerator.

File: src/test/java/org/openhds/service/AuditableServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+@WebAppConfiguration
+public abstract class AuditableServiceTest<T extends AuditableEntity,
+        U extends AbstractAuditableService, V extends AbstractTestHelper<T>>{
+
+    @Autowired
+    protected SampleDataGenerator sampleDataGenerator;
+
+    protected U service;
+
+    protected V helper;
```
Comment (benjamin-heasly) [inline src/test/java/org/openhds/service/AuditableServiceTest.java]: Love it.

**Comment by benjamin-heasly** [comment]
Looks good to me! :) :) :+1: :100:

**Comment by wolfe-lienhardt** [comment]
I have removed the helpers and added 2 abstract methods to the super tests to make valid and invalid entities.

**Comment by benjamin-heasly** [comment]
Rockin.

**Pull request merged**